### PR TITLE
[GH-229] Add Wallet and TLS specific configuration options using CLI.

### PR DESCRIPTION
**Pull request by ksaric**
https://github.com/input-output-hk/cardano-shell/issues/229

### CODE REVIEW

File: src/Cardano/Shell/Constants/CLI.hs [hunk 1]
```diff
@@ -194,3 +198,76 @@ configBlockCLIParser =
...
+    PartialTLS
+        <$> lastOption configCertificateCLIParser
+        <*> lastOption configCertificateCLIParser
+        <*> lastOption configCertificateCLIParser
+
+-- | Certificate CLI parser.
+configCertificateCLIParser :: Parser PartialCertificate
+configCertificateCLIParser =
+    PartialCertificate
+        <$> option auto
```
Comment (deepfire) [inline src/Cardano/Shell/Constants/CLI.hs]: We're not allowing defaulting here (`lastOption`)?

Comment (ksaric) [reply src/Cardano/Shell/Constants/CLI.hs]: Yep, turns out I need to fix a couple of things...

Comment (ksaric) [reply src/Cardano/Shell/Constants/CLI.hs]: Turns out those "couple of things" are an issue.
Need to rewrite the config and we need to keep the configuration above two levels deep if we want to use this "mechanism".

**Comment by ksaric** [comment]
@deepfire merged this since it's required for https://github.com/input-output-hk/cardano-shell/pull/236

**Pull request merged**